src/client/pages/LoginApp/LoginAppScreen.tsx

```diff
@@ -0,0 +1,83 @@
+import styled from '@emotion/styled'
+import { colorsV3 } from '@hedviginsurance/brand'
+import React, { useEffect } from 'react'
+import Helmet from 'react-helmet-async'
+import { FormikHelpers } from 'formik'
+import { useTextKeys } from 'utils/textKeys'
+import { Page } from 'components/utils/Page'
+import { SessionContainer } from 'containers/SessionContainer'
+import { useNorwegianBankIdAuthMutation } from 'data/graphql'
+import { useStorage } from 'utils/StorageContainer'
+import { captureSentryError } from 'utils/sentry-client'
+import { TopBar } from 'components/TopBar'
+import { LoginForm, LoginFormValue } from './components/LoginForm'
+
+const PageWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100vh;
+  padding-left: 1rem;
+  padding-right: 1rem;
+  color: ${colorsV3.gray500};
+  background-color: ${colorsV3.gray900};
+`
+
+export const LoginAppScreen: React.FC = () => {
+  const storage = useStorage()
+  const textKeys = useTextKeys()
+  const [norwegianBankIdAuthMutation] = useNorwegianBankIdAuthMutation()
+  const hasToken = Boolean(storage?.session?.getSession()?.token)
+
+  useEffect(() => {
+    if (!hasToken) {
+      return
+    }
+  }, [hasToken])
+
+  const handleSubmit = async (
+    form: LoginFormValue,
+    actions: FormikHelpers<LoginFormValue>,
+  ) => {
+    const { ssn } = form
+
+    try {
+      const result = await norwegianBankIdAuthMutation({
+        variables: { personalNumber: ssn },
+      })
+
+      if (!result) {
+        return
+      }
+
+      if (result.errors && result.errors.length > 0) {
+        actions.setErrors({ ssn: 'LOGIN_APP_GENERIC_ERROR' })
+        return
+      }
+
+      if (result.data?.norwegianBankIdAuth.redirectUrl) {
+        window.location.assign(result.data?.norwegianBankIdAuth.redirectUrl)
+      }
+    } catch (e) {
+      actions.setErrors({ ssn: 'LOGIN_APP_GENERIC_ERROR' })
+      captureSentryError(e)
+    }
```

Comment:
Don't we get the `result` as a state object from the `useNorwegianBankIdAuthMutation()`? I thought with Apollo you just execute the mutation and then use the state(s) you get back, in like a `useEffect`?
```javascript
  const [norwegianBankIdAuthMutation, { data, error }] = useNorwegianBankIdAuthMutation()
```

Reply:
Good question, I'm not sure what's the preferred way, we have it this way in a bunch of places when we trigger it when submitting, like in `AdyenCheckout`

Reply:
After looking at [Apollo's docs](https://www.apollographql.com/docs/tutorial/mutations/#apply-the-usemutation-hook) I'm pretty sure the preferred way is actually what I described - using the state object that is returned from the mutation hook. You could also use an `onCompleted` callback but that might be unnecessarily complicated.

Reply:
Ok, I guess we can change it if we feel that this is the wrong way 🙃

Reply:
After testing this I don't see a major benefit doing it this way. Since I still want to access `setErrors` in `handleSubmit`. If you guys want to amend this tomorrow, feel free and go ahead, as long as we don't delay this feature any longer please 🙏
```
  const [
    norwegianBankIdAuthMutation,
    { data },
  ] = useNorwegianBankIdAuthMutation()

  if (data?.norwegianBankIdAuth.redirectUrl) {
    window.location.assign(data.norwegianBankIdAuth.redirectUrl)
  }

  useEffect(() => {
    if (!hasToken) {
      return
    }
  }, [hasToken])

  const handleSubmit = async (
    form: LoginFormValue,
    actions: FormikHelpers<LoginFormValue>,
  ) => {
    const { ssn } = form

    try {
      await norwegianBankIdAuthMutation({
        variables: { personalNumber: ssn },
      })
    } catch (e) {
      actions.setErrors({ ssn: 'LOGIN_APP_GENERIC_ERROR' })
      captureSentryError(e)
    }
  }
```

Reply:
Yes since this is tested and verified and we want to get it out now, we can change this later if we want to

Reply:
Hm, I don't think I expressed myself clearly enough. 🙈  But never mind, we'll merge this without any changes to what you wrote initially.